convert.js

```diff
@@ -0,0 +1,361 @@
+        }
+
+        return returnCode;
+    }
+}
+
+var shapeToPathConverter = {
+    fromCircle: function (pathObject) {
+        var cx = parseInt(pathObject.attributes.cx);
+        var cy = parseInt(pathObject.attributes.cy);
+        var r = parseInt(pathObject.attributes.r);
+        var p = 'M ' + (+r + +cx) + ',' + cy + ' A ' + r + ',' + r + ' 0 0 1 ' + cx + ',' + (+r + +cy) + ' ' + r + ',' + r + ' 0 0 1 ' + (+cx - +r) + ',' + cy + ' ' + r + ',' + r + ' 0 0 1 ' + cx + ',' + (+cy - +r) + ' ' + r + ',' + r + ' 0 0 1 ' + (+cx + +r) + ',' + cy;
+
+        return p;
+    },
+    fromRect: function (pathObject) {
+        var x = parseInt(pathObject.attributes.x);
+        var y = parseInt(pathObject.attributes.y);
+        var width = parseInt(pathObject.attributes.width);
+        var height = parseInt(pathObject.attributes.height);
+
+        var p = 'M ' + x + ',' + y + ' H ' + width + ' V ' + height + ' H ' + x + ' Z';
+
+        return p;
+    }
+}
+
+function getPathData(paths) {
+    var pathData = {
+        width: 0,
+        height: 0,
+        paths: []
+    }
+
+    paths.forEach(function (pathObject) {
+        var pathString;
+        if (pathObject.name === 'circle') {
+            pathString = shapeToPathConverter.fromCircle(pathObject);
+        } else if (pathObject.name === 'rect') {
+            pathString = shapeToPathConverter.fromRect(pathObject);
+        } else {
+            pathString = pathObject.attributes.d;
+        }
+
+        path = svgUtils.path2curve(pathString)
+
+        pathData.paths.push({
+            type: 'path',
+            node: pathObject,
+            preparedPath: path
+        });
+
+        var pathBox = svgUtils.pathBBox(path);
+
+        if (pathBox.width > pathData.width) {
+            pathData.width = pathBox.width;
+        }
+
+        if (pathBox.height > pathData.height) {
+            pathData.height = pathBox.height;
+        }
+    });
+
+    return pathData;
+}
+
+function shapesToFlutterCodeConverter(shapes) {
+    lines = [];
+    var pathData = getPathData(shapes);
+    
+    pathData.paths.forEach(function (path, index) {
+        lines.push(['// Path number ' + (index + 1), 2]);
+        lines.push(['\n', 0]);
+
+        var color = colorStringToObject(getFillFromNode(path.node));
+        var opacity = path.node.attributes['fillOpacity'] == '' ? null : path.node.attributes['fillOpacity'];
+
+        if (path.type === 'path') {
+            var additionalLines = pathToFlutterConverter.convert(
+                path.preparedPath,
+                pathData.width,
+                pathData.height,
+                color,
+                opacity
+            )
+            lines = lines.concat(additionalLines);
+        }
+    });
+
+    return lines;
+}
+
+function getFillFromNode(node) {
+    var fill = '';
+
+    if (node.attributes.fill != null) {
+        fill = node.attributes.fill;
+    }
+    else {
+        fill = node.attributes.fill;
```

Comment:
Something is broken here. It does the same in both conditions. Opacity is also broken.

Reply:
I'll check that